modules/external/routes/pullrequest.ts

```diff
@@ -46,18 +45,13 @@ router.post('/', (req, res) => {
       pr = new Pullrequest(req.body);
       res.json({ status: 'Closing' });
 
-      if (pr.isMerged()) git.getIssues(pr, (issues: string[]) => bot.websocket.emit('merged', { issues }));
-
-      bot.websocket.emit('screenshot:purge', { branch: pr.branch });
-      // Loop through each skin and tell Falcon to destroy the environment
-      config.projects
-        .filter(project => project.deploy)
-        .forEach(project =>
-          bot.websocket.emit(
-            'falcon:destroy',
-            falcon.destroy(pr, project),
-          ),
-        );
+      if (pr.isMerged()) {
+        git.getIssues(pr, (issues: string[]) => bot.handleMerged(issues));
+      }
+
+      bot.purgeScreenshots(pr.branch);
+      bot.destroyAll(pr);
```

Comment:
Noticed this was the 2nd place where websockets are being manipulated directly so wanted to centralise them into the Bot and make that websocket private